File: parse-test-benchmark/src/main/java/dev/algo/parsetest/benchmark/AntlrArithmeticExprBenchmark.java
package dev.algo.parsetest.benchmark;

import dev.algo.parsetest.antlrv4.grammar.arithmetic_expr.ArithmeticExprLexer;
import dev.algo.parsetest.antlrv4.grammar.arithmetic_expr.ArithmeticExprParser;
import dev.algo.parsetest.common.Benchmark;
import org.antlr.v4.runtime.CharStreams;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.tree.ParseTree;

import java.io.IOException;
import java.nio.file.Path;

public class AntlrArithmeticExprBenchmark extends Benchmark<ParseTree> {

    public AntlrArithmeticExprBenchmark(Path folderPath) throws IOException {
        super(folderPath, ".ae");
    }

    @Override
    public ParseTree parseInput(String input) {
        ArithmeticExprLexer lexer = new ArithmeticExprLexer(CharStreams.fromString(input));
        CommonTokenStream tokens = new CommonTokenStream(lexer);
        ArithmeticExprParser parser = new ArithmeticExprParser(tokens);
        return parser.expr();
    }
}

File: parse-test-benchmark/src/main/kotlin/dev/algo/parsetest/benchmark/AntlrArithmeticExprBenchmark.kt
package dev.algo.parsetest.benchmark

import dev.algo.parsetest.antlrv4.ArithmeticExprAntlrParser
import dev.algo.parsetest.common.Benchmark
import org.antlr.v4.runtime.tree.ParseTree
import java.nio.file.Path

/**
 * Benchmark for the Antlr Parser for ArithmeticExpr
 */
class AntlrArithmeticExprBenchmark(folderPath: Path?) :
    Benchmark<ParseTree?>(folderPath, ".ae") {
    override fun parseInput(input: String): ParseTree {
        // Instantiate and parse
        return ArithmeticExprAntlrParser().parse(input)
    }
}